textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/neextractor/NamedEntityExtractorTest.java

```diff
@@ -0,0 +1,174 @@
+package edu.uci.ics.textdb.dataflow.neextractor;
+
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+
+import edu.uci.ics.textdb.api.storage.IDataReader;
+import edu.uci.ics.textdb.api.storage.IDataStore;
+import edu.uci.ics.textdb.api.storage.IDataWriter;
+import edu.uci.ics.textdb.common.constants.LuceneConstants;
+import edu.uci.ics.textdb.dataflow.neextrator.NamedEntityExtractor;
+import edu.uci.ics.textdb.dataflow.source.ScanBasedSourceOperator;
+import edu.uci.ics.textdb.dataflow.utils.TestUtils;
+import edu.uci.ics.textdb.storage.LuceneDataStore;
+import edu.uci.ics.textdb.storage.reader.LuceneDataReader;
+import edu.uci.ics.textdb.storage.writer.LuceneDataWriter;
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.Query;
+
+import org.apache.lucene.analysis.Analyzer;
+import org.junit.After;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Feng [sam0227]
+ */
+public class NamedEntityExtractorTest {
+    private NamedEntityExtractor namedEntityExtractor;
+
+    private IDataWriter dataWriter;
+    private IDataReader dataReader;
+    private IDataStore dataStore;
+
+    private Query query;
+    private Analyzer analyzer;
+
+
+    @After
+    public void cleanUp() throws Exception {
```

Comment:
In all the other test files, we are accessing the source operator where the schema and other parameters are fixed and common for all, so it can be done in setUp. In this case they are passing schema and few more as parameters. 

So one thing they can do  to maintain uniformity is to have setUp method before each test case. 

They can have @before"test" setUp method before each test case or whatever they are doing is fine i.e. sending schema and others as parameters to a method to get the source operator. It's your call Professor.